tests/http_mock.py

```diff
@@ -101,6 +101,7 @@ def request(self, uri,
                 redirections=1,
                 connection_type=None):
         resp, content = self._iterable.pop(0)
+        headers = headers.copy() if headers is not None else headers
```

Comment:
The possible assignment of `None` reads funny to me here. Why shouldn't this line be
```
headers = {} if headers is None else dict(headers)
```
?